src/win/fs.c

```diff
@@ -315,8 +328,137 @@ static int fs__readlink_handle(HANDLE handle,
     return -1;
   }
 
-  assert(target_ptr == NULL || *target_ptr == NULL);
-  return uv_utf16_to_wtf8(w_target, w_target_len, target_ptr, target_len_ptr);
+
+  if (w_target_wtf8_len_ptr)
+    *w_target_wtf8_len_ptr = uv_utf16_length_as_wtf8(w_target, w_target_len);
+
+  w_target_size = w_target_len + 1;
+  if (w_target_ptr) {
```

Comment:
Minor stylistic issue but the libuv house style is to not let pointers auto-coerce to booleans. That is...
```suggestion
  if (w_target_ptr != NULL) {
```
...here and elsewhere.